options/options.go

```diff
@@ -8,14 +8,18 @@ import (
 
 func Encode(opts interface{}) string {
 	v := url.Values{}
-	ot := reflect.TypeOf(opts)
-	ov := reflect.ValueOf(opts)
+	encodeStruct(reflect.TypeOf(opts), reflect.ValueOf(opts), &v)
+	return v.Encode()
+}
 
+func encodeStruct(ot reflect.Type, ov reflect.Value, v *url.Values) {
```

Comment:
This function is very weird and complex to me given its signature:
- it's called `encodeStruct` but there is no struct argument
- it's unclear that `v` is where the results are getting stored
- it's unclear that `ot` and `ov` are reflection of the same struct

I would suggest have to have this function be something like `func prepare(opts interface{}) url.Values` or `func extract(opts interface{}) url.Values` or similar

Reply:
Done